password_enhancements.module

```diff
@@ -522,8 +553,277 @@ function password_enhancements_legal_allowed_paths_alter(array &$allowed_paths):
+      '#field_suffix' => t('day(s) before the password expires.'),
+      '#default_value' => $policy ? $policy->getExpireWarnDays() : PasswordPolicyInterface::PASSWORD_NO_WARNING,
+      '#min' => 0,
+      '#states' => [
+        'visible' => [
+          ':input[name="password_enhancements[expire_password]"]' => ['checked' => TRUE],
+        ],
+      ],
+    ];
+
+    $form['password_enhancements']['expiryWarningMessage'] = [
+      '#type' => 'textarea',
+      '#title' => t('Expiry warning message'),
+      // @TODO: #item_list
+      '#description' => t("This message will be presented for the user if the user's password is about to expire.<br>Available placeholders: <ul><li>@url: URL for the user edit page</li><li>@date_time: the date and time when the password will expire</li></ul>."),
+      '#default_value' => $policy ? $policy->getExpiryWarningMessage() : 'Your password will expire on @date_time, please <a href="@url">change your password</a> before it expires to prevent any potential data loss.',
+      '#states' => [
+        'invisible' => [
+          ':input[name="password_enhancements[expire_warn_before_days]"]' => ['value' => 0],
+        ],
+      ],
+    ];
+
+    // Only allow adding constraints when the policy is already set up.
+    if ($policy) {
+      // Build the list of existing password constraints for this password
+      // policy.
+      $form['password_enhancements']['constraints'] = [
+        '#type' => 'table',
+        '#header' => [
+          t('Constraint'),
+          t('Required'),
+          t('Summary'),
+          t('Operations'),
+        ],
+        '#attributes' => [
+          'id' => 'password-policy-constraints',
+        ],
+        '#empty' => t('There are currently no constraints in this policy. Add one by selecting an option below.'),
+        // Render constraints below parent elements.
+        '#weight' => 5,
+      ];
+      $old_unique_constraints = [];
+      /** @var \Drupal\password_enhancements\PasswordConstraintInterface $constraint */
+      foreach ($policy->getConstraints() as $constraint) {
+        if ($constraint->getPluginDefinition()['unique']) {
+          $old_unique_constraints[$constraint->getPluginId()] = $constraint->getPluginId();
+        }
+        $key = $constraint->getUuid();
+        $form['password_enhancements']['constraints'][$key]['constraint'] = [
+          '#tree' => FALSE,
+          'data' => [
+            'name' => [
+              '#plain_text' => $constraint->name(),
+            ],
+          ],
+        ];
+
+        $form['password_enhancements']['constraints'][$key]['required'] = [
+          '#plain_text' => $constraint->isRequired() ? t('Yes') : t('No'),
+        ];
+
+        if ($summary = $constraint->getSummary()) {
+          $form['password_enhancements']['constraints'][$key]['summary'] = $summary;
+        }
+
+        $links = [
```

Comment:
As all the PwEnh-related strings are stored as thirdPartySettings on the role, the only way to translate them is translating the role. Because of that, I don't think a translate link is needed for the constraint instances.